src/Bicep.Core/Emit/PlaceholderParametersBicepParamWriter.cs

```diff
@@ -79,26 +79,39 @@ private SyntaxBase GetValueForParameter(ParameterDeclarationSyntax syntax)
                 return defaultValue;
             }
 
+            if (syntax.Type is ObjectTypeSyntax objectTypeSyntax)
+            {
+                var value = objectTypeSyntax.Properties.Select(e =>
+                {
+                    var identifierName = (e.Key as IdentifierSyntax)?.IdentifierName ?? "";
+
+                    return ((e.Value as TypeVariableAccessSyntax)?.Name.IdentifierName) switch
+                    {
+                        "int" => SyntaxFactory.CreateObjectProperty(identifierName, SyntaxFactory.CreateIntegerLiteral(0)),
+                        "bool" => SyntaxFactory.CreateObjectProperty(identifierName, SyntaxFactory.CreateBooleanLiteral(false)),
+                        "array" => SyntaxFactory.CreateObjectProperty(identifierName, SyntaxFactory.CreateArray([])),
+                        "object" => SyntaxFactory.CreateObjectProperty(identifierName, SyntaxFactory.CreateObject([])),
+                        _ => SyntaxFactory.CreateObjectProperty(identifierName, SyntaxFactory.CreateStringLiteral(string.Empty)),
+                    };
+                });
+
+                return SyntaxFactory.CreateObject(value);
+            }
+
             if (syntax.Type is TypeVariableAccessSyntax variableAccessSyntax)
             {
                 var allowedDecorator = syntax.Decorators.Where(e => e.Expression is FunctionCallSyntax functionCallSyntax && functionCallSyntax.Name.IdentifierName == "allowed").Select(e => e.Arguments).FirstOrDefault();
 
                 var allowedDecoratorFirstItem = (allowedDecorator?.First().Expression as ArraySyntax)?.Items.First().Value;
 
-                switch (variableAccessSyntax.Name.IdentifierName)
+                return variableAccessSyntax.Name.IdentifierName switch
```

Comment:
I think this logic needs to be made recursive - I think you would need to:

* Create a new method `private static SyntaxBase GetSyntaxForType(TypeSymbol type)` which generates placeholder syntax for different types (e.g. ObjectType, StringType, etc)
* Obtain the type of `syntax.Type` from the semantic model, and use this for syntax generation.

Reply:
with the latest commit inner objects are handled 👍

ensure it, I added a few more tests.

for example the following param definition;

```bicep
param bab {{ bar: string, baz: {{ foo: int }} }} = {{ bar: 'hello', baz: {{ foo: 42 }} }}
```

generate correct value

```bicep
param bat = {
  bar: ''
  baz: {
    foo: {
      bar: {
        baz: ''
      }
    }
  }
}
```